feat(linter): add fixer for @typescript-eslint/consistent-type-imports

mysteryven opened the pull request:
This rule's fixer is complicated, I do line by line copy as possible.

https://github.com/typescript-eslint/typescript-eslint/blob/7b13dae347075d467ea95d0f986b38c3048906a2/packages/eslint-plugin/src/rules/consistent-type-imports.ts

mysteryven commented:
* **#3984** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/3984?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @mysteryven and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="11px" height="11px"/> <b>Graphite</b></a>
  <!-- Current dependencies on/for this PR: -->

rzvxa commented on crates/oxc_linter/src/rules/typescript/consistent_type_imports.rs:
```diff
@@ -1532,1053 +2179,1096 @@ fn test() {
+            let foo: Foo;
+                      ",
+            Some(
+                serde_json::json!([            { "prefer": "type-imports", "fixStyle": "inline-type-imports" },          ]),
```
I'd like to review this rule but as you've mentioned it is quite complicated, So depending on how my day goes it might take me a few hours to a day before I'm done with it. Don't let my slow review stop you from merging it if anyone else has the time to leave a review before me.

----------

On another note:
There are some formatting issues in the test cases, You may want to expand your tabs as spaces(that's usually why my code formatting is broken because I don't expand tabs by default and GitHub uses 8 tab size by default as opposed to 4 which I use locally).

mysteryven replied:
Now the test cases in `fix` are formatted. 

> how my day goes it might take me a few hours to a day before I'm done with it.

Good luck to you =.=, and looking forward to your suggestion.

Boshen commented:
This is a big ask, but can you test this in https://github.com/oxc-project/oxlint-ecosystem-ci manually? 

The repos are cloned locally, you can check `git diff` after the fix.

More over, I think we should add idempotence test to the fixer: run fix, then run fix agin on the fixed code, the result should be the same.

Boshen commented:
Merge first, integrate with ecosystem ci later.

mysteryven committed:
feat(linter): add fixer for @typescript-eslint/consistent-type-imports (#3984)

This rule's fixer is complicated, I do line by line copy as possible.

https://github.com/typescript-eslint/typescript-eslint/blob/7b13dae347075d467ea95d0f986b38c3048906a2/packages/eslint-plugin/src/rules/consistent-type-imports.ts
